Libraries/CustomComponents/Navigator/Navigator.js

```diff
@@ -1043,6 +1043,14 @@ var Navigator = React.createClass({
     return this.state.routeStack.slice();
   },
 
+  getCurrentRouteIndex: function() {
+    return this.state.presentedIndex;
+  },
+
+  getCurrentRoute: function() {
+    return {...this.state.routeStack[this.getCurrentRouteIndex()]};
```

Comment:
I really hope that we could stop adding new getters to expose the internal state of the navigator. Once we move toward the new controlled-view model, it won't make much sense to let the component hold the state.